### PR TITLE
fix(zacas): fix check exception for amocas.

### PR DESCRIPTION
**Pull request by NewPaulWalker**

### CODE REVIEW

File: src/isa/riscv64/instr/rva/amo.c [hunk 1]
```diff
@@ -90,13 +99,16 @@ def_rtl(amo_slow_path, rtlreg_t *dest, const rtlreg_t *src1, const rtlreg_t *src
...
-    if (width == 16 && ((rd % 2 == 1) || (rs2 % 2 == 1))) {
-      longjmp_exception(EX_II);
-    }
     cpu.amo = true;
+    // check store behavior before actually load or store
+    uint64_t paddr = *dsrc1;
+    if (isa_mmu_check(*dsrc1, width, MEM_TYPE_WRITE) == MMU_TRANSLATE) {
+      paddr = isa_mmu_translate(*dsrc1, width, MEM_TYPE_WRITE);
+    }
+    if (!in_pmem(paddr) || !isa_pmp_check_permission(paddr, width, MEM_TYPE_WRITE, cpu.mode)) {
```
Comment (huxuan0307) [inline src/isa/riscv64/instr/rva/amo.c:108]: cpu.mode should be MPRV?mstatus.MPP:cpu.mode

**Review by huxuan0307 — CHANGES_REQUESTED** [review]
Wrong pmp_check_permission argument needs modification

File: src/isa/riscv64/instr/rva/amo.c (hunk 1 above)
Comment (NewPaulWalker) [reply src/isa/riscv64/instr/rva/amo.c:108]: The memory access mode has been obtained in `isa_pmp_check_permission`.

https://github.com/OpenXiangShan/NEMU/blob/master/src/isa/riscv64/system/mmu.c#L1017
>   mode = (out_mode == MODE_M) ? (mstatus->mprv && !ifetch ? mstatus->mpp : cpu.mode) : out_mode;

**Review by huxuan0307 — APPROVED** [review]

**Comment by huxuan0307** [comment]
> Wrong pmp_check_permission argument needs modification

My misunderstanding.

**Pull request merged**